internal/parser/translate.go

```diff
@@ -327,84 +327,155 @@ func fromIngressV1(log logrus.FieldLogger, ingressList []*networkingv1.Ingress)
+			// Therefore we treat each match rule as a separate Kong Route, so we iterate through
+			// all matches to determine all the routes that will be needed for the services.
+			routes := make([]kongstate.Route, 0, len(rule.Matches))
+			for matchNumber, match := range rule.Matches {
+				// determine the name of the route, identify it as a route that belongs
+				// to a Kubernetes HTTPRoute object.
+				routeName := kong.String(fmt.Sprintf(
+					"httproute.%s.%s.%d",
+					httproute.Namespace,
+					httproute.Name,
+					matchNumber,
+				))
+
+				// TODO: implement query param matches
+				if len(match.QueryParams) > 0 {
+					log.Errorf("regular expression path matches are not yet supported, HTTPRoute %s/%s can't be routed for match %+v", httproute.Namespace, httproute.Name, match)
+					continue
+				}
+
+				// TODO: implement regex path matches
+				if *match.Path.Type == gatewayv1alpha2.PathMatchRegularExpression {
+					log.Errorf("regular expression path matches are not yet supported, HTTPRoute %s/%s can't be routed for match %+v", httproute.Namespace, httproute.Name, match)
+					continue
+				}
+
+				// build the route object using the method and pathing information
+				r := kongstate.Route{
+					Ingress: objectInfo,
+					Route: kong.Route{
+						Name:         routeName,
+						Protocols:    kong.StringSlice("http", "https"),
+						PreserveHost: kong.Bool(true),
+						Hosts:        hostnames,
+					},
+				}
+				log.Debugf("generated route %s for HTTPRoute %s/%s", routeName, httproute.Namespace, httproute.Name)
+
+				// configuration path matching information about the route if paths
+				// matching was defined.
+				if match.Path != nil {
+					// determine the path match values
+					r.Route.Paths = []*string{match.Path.Value}
+
+					// determine whether path stripping needs to be enabled
+					r.Route.StripPath = kong.Bool(match.Path.Type == nil || *match.Path.Type == gatewayv1alpha2.PathMatchPathPrefix)
+				}
+
+				// configure method matching information about the route if method
+				// matching was defined.
+				if match.Method != nil {
+					r.Route.Methods = append(r.Route.Methods, kong.String(string(*match.Method)))
+				}
+
+				// convert header matching to from HTTPRoute to Route format
+				invalidHeaders := false
+				headers := make(map[string][]string)
+				for _, header := range match.Headers {
+					// TODO: implement regex header matching
+					if *header.Type == gatewayv1alpha2.HeaderMatchRegularExpression {
+						log.Errorf("regular expression header matches are not yet supported, HTTPRoute %s/%s can't be routed for match %+v", httproute.Namespace, httproute.Name, match)
+						invalidHeaders = true
+					}
+
+					// store the header matching information
+					values := strings.Split(header.Value, ",")
+					headers[string(header.Name)] = values
+				}
+				if invalidHeaders {
+					continue
+				}
+				if len(headers) > 0 {
```

Comment:
is this `if` necessary?

Reply:
No, it's now removed in f19290b3e5674e9fd8247165d101d70c952ff537